### PR TITLE
feature: Video Message Feature for Embedded Chat

### LINKED ISSUE #436: [ Feature ] Video Message Feature for Embedded Chat
As an EmbeddedChat developer

**I need to**:
Implement a video message feature within the Embedded Chat interface.

**So That**:
Users can send and receive video messages seamlessly, enhancing the communication experience within the chat.

**Acceptance Criteria**

- [x] Add a video message icon to the chat input formatting toolbar.
- [x] Implement video recording capability.
- [x] Allow users to preview recorded videos before sending.

### PR DESCRIPTION
**Pull request by shivang-16**
### Acceptance Criteria fulfillment

- [x]  Added a video message icon to the chat input formatting toolbar.
- [x]  Implement video recording capability.
- [x]  Allow users to view recording videos while recording.

### Video/Screenshots

https://github.com/RocketChat/EmbeddedChat/assets/125182653/af6cc52d-c22b-4e43-a7c3-a02b4559760d

### CODE REVIEW

File: packages/react/src/components/Icon/icons/index.js [hunk 1]
```diff
@@ -46,6 +48,8 @@ const icons = {
   computer: Computer,
   cross: Cross,
   mic: Mic,
+  videoRecorder: VideoRecorder,
```
Comment (abhinavkrin) [inline packages/react/src/components/Icon/icons/index.js:51]: use case `video-recorder`

File: packages/react/package.json [hunk 2]
```diff
@@ -104,6 +104,7 @@
     "js-cookie": "^3.0.5",
     "normalize.css": "^8.0.1",
     "prop-types": "^15.8.1",
+    "react-modal": "^3.16.1",
```
Comment (abhinavkrin) [inline packages/react/package.json:107]: use our modal component. modify if needed.

**Review by abhinavkrin — CHANGES_REQUESTED** [review]

File: packages/react/package.json (hunk 2 above)
Comment (shivang-16) [reply packages/react/package.json]: done

File: packages/react/src/components/Icon/icons/index.js (hunk 1 above)
Comment (shivang-16) [reply packages/react/src/components/Icon/icons/index.js]: done

**Comment by abhinavkrin** [comment]
Please Share video of the latest behaviour.

**Comment by shivang-16** [comment]
Here is the video of latest change

https://github.com/RocketChat/EmbeddedChat/assets/125182653/49fb28c0-9fa0-4832-91d7-933efbac1e8f

**Review by abhinavkrin — APPROVED** [review]

**Pull request merged**